internal/rpcserver/interface.go

```diff
@@ -134,3 +138,300 @@ type SyncManager interface {
+	// wire.MaxBlockHeadersPerMsg headers.
+	//
+	// In addition, there are two special cases:
+	//
+	// - When no locators are provided, the stop hash is treated as a request for
+	//   that header, so it will either return the header for the stop hash itself
+	//   if it is known, or nil if it is unknown
+	// - When locators are provided, but none of them are known, headers starting
+	//   after the genesis block will be returned
+	//
+	// This function is safe for concurrent access.
+	LocateHeaders(locator blockchain.BlockLocator, hashStop *chainhash.Hash) []wire.BlockHeader
+
+	// LotteryDataForBlock returns lottery data for a given block in the block
+	// chain, including side chain blocks.
+	//
+	// It is safe for concurrent access.
+	LotteryDataForBlock(hash *chainhash.Hash) ([]chainhash.Hash, int, [6]byte, error)
+
+	// MainChainHasBlock returns whether or not the block with the given hash is in
+	// the main chain.
+	//
+	// This function is safe for concurrent access.
+	MainChainHasBlock(hash *chainhash.Hash) bool
+
+	// MaxBlockSize returns the maximum permitted block size for the block AFTER
+	// the end of the current best chain.
+	//
+	// This function is safe for concurrent access.
+	MaxBlockSize() (int64, error)
+
+	// MissedTickets returns all currently missed tickets from the stake database.
+	//
+	// This function is safe for concurrent access.
+	MissedTickets() ([]chainhash.Hash, error)
+
+	// NextThresholdState returns the current rule change threshold state of the
+	// given deployment ID for the block AFTER the provided block hash.
+	//
+	// This function is safe for concurrent access.
+	NextThresholdState(hash *chainhash.Hash, version uint32, deploymentID string) (blockchain.ThresholdStateTuple, error)
+
+	// StateLastChangedHeight returns the height at which the provided consensus
+	// deployment agenda last changed state.  Note that, unlike the ThresholdState
+	// function, this function returns the information as of the passed block hash.
+	//
+	// This function is safe for concurrent access.
+	StateLastChangedHeight(hash *chainhash.Hash, version uint32, deploymentID string) (int64, error)
+
+	// TicketPoolValue returns the current value of all the locked funds in the
+	// ticket pool.
+	//
+	// This function is safe for concurrent access.  All live tickets are at least
+	// 256 blocks deep on mainnet, so the UTXO set should generally always have
+	// the asked for transactions.
+	TicketPoolValue() (dcrutil.Amount, error)
+
+	// TicketsWithAddress returns a slice of ticket hashes that are currently live
+	// corresponding to the given address.
+	//
+	// This function is safe for concurrent access.
+	TicketsWithAddress(address dcrutil.Address) ([]chainhash.Hash, error)
+
+	// TipGeneration returns the entire generation of blocks stemming from the
+	// parent of the current tip.
+	//
+	// The function is safe for concurrent access.
+	TipGeneration() ([]chainhash.Hash, error)
+}
```

Comment:
This seems pretty verbose since rpcserver relies on quite a few blockchain methods, but also seems like the clearest (and more explicit) way to outline what rpcserver currently uses while also allowing for mocking - open to suggestions if anyone has a better idea

Reply:
I meant more just defining that many methods in the interface that align one-to-one with the blockchain methods - vs potentially breaking this down into smaller interfaces and/or consolidating some of the interfaces to be more geared for what rpcserver needs.

One issue with not commenting the interface would be that most editors have plugins that allow for peeking/viewing method comments on hover, so you would lose that ability

Reply:
I gotcha. imo this is about testing the handlers, and it seems you found a bug, so goal achieved. Optimization can come later, if needed.

Good point about the editors. That's something I hadn't thought of. Let's keep the comments then.